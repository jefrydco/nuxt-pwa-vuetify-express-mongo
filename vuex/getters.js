export default {
  isNotif: state => state.notif.text !== null && state.notif.color !== null
};
