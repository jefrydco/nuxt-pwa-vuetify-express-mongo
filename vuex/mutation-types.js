export default {
  SET_NOTIF: "SET_NOTIF"
};
