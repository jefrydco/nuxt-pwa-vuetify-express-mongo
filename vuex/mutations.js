import types from "./mutation-types";

export default {
  [types.SET_NOTIF]: (state, { text, color }) => {
    state.notif = {
      text,
      color
    };
  }
};
