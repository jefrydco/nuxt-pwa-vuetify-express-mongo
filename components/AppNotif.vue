<template>
  <v-snackbar
    :value="isNotif"
    :color="notif.color"
    right=""
    @input="onCloseButtonClicked"
  >
    {{ notif.text }}
    <v-btn dark flat @click="onCloseButtonClicked"> Close </v-btn>
  </v-snackbar>
</template>

<script>
import { mapGetters } from "vuex";
import notif from "~/mixins/notif";

export default {
  mixins: [notif],
  computed: {
    ...mapGetters(["isNotif"])
  },
  methods: {
    onCloseButtonClicked() {
      this.setNotif();
    }
  }
};
</script>
