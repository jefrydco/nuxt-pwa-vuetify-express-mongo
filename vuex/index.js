import getters from "./getters";
import mutations from "./mutations";
import actions from "./actions";

const state = () => ({
  notif: {
    text: null,
    color: null
  }
});

export default {
  state,
  getters,
  mutations,
  actions
};
