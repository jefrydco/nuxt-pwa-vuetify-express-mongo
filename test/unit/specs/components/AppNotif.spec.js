import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Vuetify from "vuetify";
import VeeValidate from "vee-validate";
import moment from "moment";
import axios from "axios";

import AppNotif from "~/components/AppNotif";

import createRepo from "~/helpers/api";
import vuex from "~/vuex";

const localVue = createLocalVue();
const createApi = createRepo(axios);
const apiList = {
  products: createApi("products")
};

localVue.use(Vuex);
localVue.use(Vuetify);
localVue.use(VeeValidate);

describe("AppNotif component", () => {
  let store;

  beforeAll(() => {
    store = new Vuex.Store({
      ...vuex
    });
  });

  it("renders to match snapshot", () => {
    const wrapper = shallowMount(AppNotif, {
      localVue,
      store,
      mocks: {
        $moment: moment,
        $api: apiList
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
