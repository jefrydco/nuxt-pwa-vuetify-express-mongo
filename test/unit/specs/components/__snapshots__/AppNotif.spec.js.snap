// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNotif component renders to match snapshot 1`] = `
<v-snackbar-stub timeout="6000" right="">

  <v-btn-stub activeclass="v-btn--active" flat="true" tag="button" type="button" dark=""> Close </v-btn-stub>
</v-snackbar-stub>
`;
